R/fcm.R

```diff
@@ -210,41 +210,43 @@ fcm.tokens_xptr <- function(x, context = c("document", "window"),
     tri <- check_logical(tri)
 
     attrs <- attributes(x)
-    if (ordered)
+    if (ordered) {
+        if (tri)
+            warning("cannot return only upper triangle when ordered = TRUE")
         tri <- FALSE
-    if (context == "document") {
-        result <- fcm(dfm(x, tolower = FALSE, verbose = FALSE), count = count, tri = tri)
-    } else {
-        if (count == "weighted") {
-            if (!is.null(weights)) {
-                weights <- check_double(weights, max_len = Inf)
-                if (length(weights) != window)
-                    stop("The length of weights must be equal to the window size")
-            } else {
-                weights <- 1 / seq_len(window)
-            }
+    }
+    if (context == "document")
+        window <- max(ntoken(x))
```

Comment:
See my comment [here](https://github.com/quanteda/quanteda/issues/2458#issuecomment-2893300183). Setting   this to the max length is not the same as context = "document" since it still allows different behaviour when `ordered = TRUE`. The idea is not just one of length, it's one of document cooccurrence that has no concept of order. That's why in my suggestion, we remove context = "document" entirely from tokens, and suggest that users create a dfm if that's what they want.

If we want to provide an option as in this line, we could either provide `-1` to mean `max(ntoken(x))` or we could just suggest that users input this themselves for window or some other effectively equivalently large value.

Reply:
Let's remove `context == "document"`. This might be even more confusing.